### ISSUE TITLE
ProjectMergeRequest.merge returns GitlabMRClosedError on merge conflicts

### ISSUE THREAD

**Issue by gzur** [python-gitlab/python-gitlab#1378]
If [`ProjectMergeRequest.merge`](https://github.com/python-gitlab/python-gitlab/blob/0b67ca29d2cc6177e330b91519fdf54b05621769/gitlab/v4/objects/merge_requests.py#L299) fails due to **merge-conflicts**, it raises a `GitlabMRClosedError`:

with the message:
```
Caught <class 'gitlab.exceptions.GitlabMRClosedError'> error: 406: Branch cannot be merged
```
This is highly confusing, since the exception implies that the Merge Request has been Closed, which is not necessarily the case.

We've been running into this in our pipelines and it caused quite a bit of confusion until we looked at the code.

I suggest either trying to figure out **why** the request failed, or at least raise a more generic/representative exception, since it seems to raise `GitlabMRClosedError` in the case of any HTTP Error:
https://github.com/python-gitlab/python-gitlab/blob/0b67ca29d2cc6177e330b91519fdf54b05621769/gitlab/v4/objects/merge_requests.py#L299-L300

**Comment by nejch** [comment]
See also https://github.com/python-gitlab/python-gitlab/issues/1198#issuecomment-703150606 for more info on this, seems like this is a duplicate and that should be reopened. I'll update this in a bit.

TLDR GitlabMRClosedError needs to be deprecated and there is really no way to tell what the exact reason for the 405 response is.
